bindings/rust/bench/benches/throughput.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use bench::{
+    CipherSuite::{self, *},
+    CryptoConfig,
+    ECGroup::X25519,
+    OpenSslHarness, RustlsHarness, S2NHarness, TlsBenchHarness,
+};
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement::WallTime, BatchSize, BenchmarkGroup,
+    Criterion, Throughput,
+};
+use std::any::type_name;
+
+pub fn bench_throughput_cipher_suite(c: &mut Criterion) {
+    let mut shared_buf = [0u8; 100000];
+
+    fn bench_throughput_for_library<T: TlsBenchHarness>(
+        bench_group: &mut BenchmarkGroup<WallTime>,
+        shared_buf: &mut [u8],
+        cipher_suite: &CipherSuite,
+    ) {
+        bench_group.bench_function(type_name::<T>(), |b| {
+            b.iter_batched_ref(
+                || {
+                    let mut harness = T::new(&CryptoConfig {
+                        cipher_suite: *cipher_suite,
+                        ec_group: X25519,
+                    })
+                    .unwrap();
+                    harness.handshake().unwrap();
+                    harness
+                },
+                |harness| harness.round_trip_transfer(black_box(shared_buf)).unwrap(),
```

Comment:
Why is black box necessary? I think it has something to do with preventing the compiler from optimizing out certain details, but that doesn't feel like it should be necessary here.

Reply:
I've tried it without `black_box()` and it doesn't change any results, but since theoretically everything we're benching is deterministic, I'm scared that one day maybe the compiler will optimize it away and we won't immediately realize why throughput is so high.

Reply:
`black_box()` is only for inputs to a function being benched, such as `round_trip_transfer()`. Because `handshake()`, the other function benched, doesn't take in any inputs, there are no inputs to optimize away.

Reply:
Hmm, what would the compiler be optimizing away? I think Madeline's questions are very relevant, and if we leave it in the explicit expectation of future engineers is going to be that it _needs_ to be included.

Traditionally I've seen it used for thinks like `black_box(2 + 3)` where it prevent the compiler from doing compile-time evaluation of the addition. Here it doesn't seem like there is necessarily anything to optimize away, so I'd prefer to remove it.

Reply:
I agree that leaving it in will set an incorrect expectation. I'm also not that familiar with how compiler optimization works, and it isn't a simple operation like addition, so it does make sense that the compiler won't optimize it away. I'll remove the `black_box()`.